src/App.tsx

```diff
@@ -8,37 +8,92 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { getUsers } from './api/api';
+import { User } from './types/User';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isLoadingPosts, setIsLoadingPosts] = useState(false);
+  const [postsError, setPostsError] = useState(false);
+  const [addError, setAddError] = useState(false);
+  const [deleteError, setDeleteError] = useState(false);
+  const [usersError, setUsersError] = useState(false);
+
+  const noPostsWarning
+    = !postsError && !posts.length && selectedUser && !isLoadingPosts;
+
+  useEffect(() => {
+    getUsers()
+      .then(setUsers)
+      .catch(setUsersError);
+  }, []);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  setPosts={setPosts}
+                  setIsLoadingPosts={setIsLoadingPosts}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                  setPostsError={setPostsError}
+                  setSelectedPost={setSelectedPost}
```

Comment:
do not pass all functions there
create function `handleSelectUser` where you do everything you need in `App.tsx` and pass it to `UserSelector`